### PR TITLE
added concurrent cleanup before runner pod is deleted

### LINKED ISSUE #6: When pods are deleted, cleanup linked pods
IF we are running in k8s mode, delete all pods in the appropriate namespace with label `runner-pod:{RUNNERPODNAME}` when that pod is deleted.

We likely want to do this during pod downsizing/reconciliation job

### PR DESCRIPTION
**Pull request by nikola-jokic**
Closes #6

### CODE REVIEW

File: controllers/runner_controller.go [hunk 1]
```diff
@@ -112,6 +114,34 @@ func (r *RunnerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 			// Pod was not found
 			return r.processRunnerDeletion(runner, ctx, log, nil)
 		}
+		// remove runner-linked pods
+		var runnerLinkedPodList corev1.PodList
+		r.List(ctx, &runnerLinkedPodList, client.MatchingLabels(
```
Comment (thboop) [inline controllers/runner_controller.go:119]: do we need to check the state? aka only try to delete pods that aren't already deleting?

Comment (thboop) [reply controllers/runner_controller.go:119]: Do we also need a retry behavior for this, like we have for getting the runner pod above?

Comment (nikola-jokic) [reply controllers/runner_controller.go:119]: Thanks, I will add that as well!

**Review by thboop — APPROVED** [review]
very clean 🚢

**Comment by nikola-jokic** [comment]
Thank you :relaxed:

**Pull request merged**